Makefile

```
@@ -291,11 +291,11 @@ endif
 ifneq ($(wildcard $(XSMM_DIR)/lib/libxsmm.*),)
   $(libceeds) : LDFLAGS += -L$(XSMM_DIR)/lib -Wl,-rpath,$(abspath $(XSMM_DIR)/lib)
   $(libceeds) : LDLIBS += -lxsmm -ldl
-  MKL ?= 0
-  ifneq (0,$(MKL))
-    BLAS_LIB = -Wl,--no-as-needed -lmkl_intel_lp64 -lmkl_sequential -lmkl_core -lpthread -lm -ldl
-  else
+  MKL ?=
+  ifeq (,$(MKL)$(MKLROOT))
     BLAS_LIB = -lblas
+  else
+    BLAS_LIB = -L$(MKLROOT)/lib/intel64 -Wl,-rpath,$(abspath $(MKLROOT)/lib/intel64),--no-as-needed -lmkl_intel_lp64 -lmkl_sequential -lmkl_core -lpthread -lm -ldl
```

If MKLROOT was empty, then you could be passing `-L/lib/intel64`, which may not exist, and thus produce errors. Perhaps something like this?
```suggestion
    BLAS_LIB = $(if $(MKLROOT),-L$(MKLROOT)/lib/intel64 -Wl,-rpath,$(MKLROOT)/lib/intel64) -Wl,--no-as-needed -lmkl_intel_lp64 -lmkl_sequential -lmkl_core -lpthread -lm -ldl
```

---

All set